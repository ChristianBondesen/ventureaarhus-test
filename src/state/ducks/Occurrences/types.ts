export const SET_OCCURRENCES = 'user/SET_OCCURRENCES';

export type SET_OCCURRENCES = typeof SET_OCCURRENCES;

export interface IEvent {
  '@context': string;
  '@id': string;
  '@type': string;
  updatedAt: string;
  isPublished: boolean;
  occurrences: IOccurrence[];
  ticketPurchaseUrl?: any;
  eventUrl?: any;
  excerpt: string;
  organizer: IOrganizer;
  tags: string[];
  customTags: string[];
  description: string;
  image: string;
  name: string;
  url: string;
  videoUrl?: any;
  langcode: string;
}

export interface IOrganizer {
  '@id': string;
  '@type': string;
  name: string;
  email: string;
  url: string;
}

export interface IOccurrence {
  '@id': string;
  '@type': string;
  event: string;
  startDate: Date;
  endDate: Date;
  place: IPlace;
  room?: any;
  ticketPriceRange: string;
  eventStatusText: string;
}

export interface IPlace {
  '@id': string;
  '@type': string;
  telephone?: any;
  email: string;
  logo?: any;
  disabilityAccess?: any;
  latitude?: number;
  longitude?: number;
  tags: string[];
  description: string;
  image: string;
  name: string;
  url: string;
  videoUrl?: any;
  langcode: string;
}
