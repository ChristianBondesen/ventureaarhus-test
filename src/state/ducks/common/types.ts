export const CHANGE_CONNECTION_STATUS = 'common/CHANGE_CONNECTION_STATUS';
export type CHANGE_CONNECTION_STATUS = typeof CHANGE_CONNECTION_STATUS;

export const INCREMENT_NETWORK_CALLS = 'common/INCREMENT_NETWORK_CALLS';
export type INCREMENT_NETWORK_CALLS = typeof INCREMENT_NETWORK_CALLS;
