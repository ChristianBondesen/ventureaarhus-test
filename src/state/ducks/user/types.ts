export const SET_USER = 'user/SET_USER'; // Adds user to the store

export type SET_USER = typeof SET_USER;

export interface IUser {
  id: number;
  name: string;
  thumbnail: string;
  likes: IEvent[];
  participate: IEvent[];
  follows: IEvent[];
  // more to come
}

export interface IEvent {
  id: number;
  name: string;
  date: Date;
  ticketUrl: string;
  // more to come
}
