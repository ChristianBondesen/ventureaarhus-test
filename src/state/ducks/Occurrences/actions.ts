import { IFluxStandardAction } from '../FluxStandardAction';
import * as types from './types';

// Interface for your actions -->
export interface ISetOccurrences extends IFluxStandardAction {
  type: types.SET_OCCURRENCES;
  payload: types.IOccurrence[];
}

export interface ISetOccurrencesTags extends IFluxStandardAction {
  type: types.SET_OCCURRENCES_TAGS;
  payload: types.ITags[];
}

export interface IUpdateRecommendedOccurrences extends IFluxStandardAction {
  type: types.UPDATE_RECOMMENDED_OCCURRENCES;
  payload: types.IOccurrence[];
}

// State of your Actions -->
// (Import for the reducer)
export type OccurrenceStateAction =
  | ISetOccurrences
  | ISetOccurrencesTags
  | IUpdateRecommendedOccurrences;

// Actual action implementing the former defined interface -->
export const setOccurrenceList = (
  occurrences: types.IOccurrence[]
): ISetOccurrences => ({
  type: types.SET_OCCURRENCES,
  payload: occurrences,
});

export const setOccurrencesTags = (
  tags: types.ITags[]
): ISetOccurrencesTags => ({
  type: types.SET_OCCURRENCES_TAGS,
  payload: tags,
});

export const updateRecommendedOccurrences = (
  recOccurs: types.IOccurrence[]
): IUpdateRecommendedOccurrences => ({
  type: types.UPDATE_RECOMMENDED_OCCURRENCES,
  payload: recOccurs,
});

// Interface of all your Actions -->
export interface IOccurrenceActions {
  setOccurrenceList(occurrences: types.IOccurrence[]);
  setOccurrencesTags(tags: types.ITags[]);
  updateRecommendedOccurrences(recOccurs: types.IOccurrence[]);
}

// Export the actions -->
export default {
  setOccurrenceList,
  setOccurrencesTags,
  updateRecommendedOccurrences,
};
