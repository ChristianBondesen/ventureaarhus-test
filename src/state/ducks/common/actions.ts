import { IFluxStandardAction } from '../FluxStandardAction';
import * as types from './types';

export interface IChangeConnectionStatus extends IFluxStandardAction {
  type: types.CHANGE_CONNECTION_STATUS;
  payload: boolean;
}
export interface IIncrementNetworkCalls extends IFluxStandardAction {
  type: types.INCREMENT_NETWORK_CALLS;
}

export type CommonStateAction =
  | IChangeConnectionStatus
  | IIncrementNetworkCalls;

export const setConnectionStatus = (
  connected: boolean
): IChangeConnectionStatus => ({
  type: types.CHANGE_CONNECTION_STATUS,
  payload: connected,
});

export const incrementNetworkCalls = (): IIncrementNetworkCalls => ({
  type: types.INCREMENT_NETWORK_CALLS,
});

export interface ICommonActions {
  setConnectionStatus(connected: boolean);
  incrementNetworkCalls();
}

export default {
  setConnectionStatus,
  incrementNetworkCalls,
};
