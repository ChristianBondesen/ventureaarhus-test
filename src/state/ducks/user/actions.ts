import { IFluxStandardAction } from '../FluxStandardAction';
import * as types from './types';

export interface ISetUser extends IFluxStandardAction {
  type: types.SET_USER;
  payload: types.IUser;
}

export type UserStateAction = ISetUser;

export const setUser = (user: types.IUser): ISetUser => ({
  type: types.SET_USER,
  payload: user,
});

export interface IUserActions {
  setUser(user: types.IUser);
}

export default {
  setUser,
};
