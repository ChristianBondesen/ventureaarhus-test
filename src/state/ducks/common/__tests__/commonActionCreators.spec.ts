import * as actions from './../actions';
import * as types from './../types';

const TAG = 'commonActionCreators.spec.js';

describe('Common action creators', () => {
  it('should properly create a change action for connection status', () => {
    const expectedActionForFalse = {
      type: types.CHANGE_CONNECTION_STATUS,
      payload: false,
    };
    const expectedActionForTrue = {
      type: types.CHANGE_CONNECTION_STATUS,
      payload: true,
    };
    expect(actions.setConnectionStatus(false)).toEqual(expectedActionForFalse);
    expect(actions.setConnectionStatus(true)).toEqual(expectedActionForTrue);
  });

  it('should properly create a change action for connection status using snapshot', () => {
    expect(actions.setConnectionStatus(false)).toMatchSnapshot();
  });
});
