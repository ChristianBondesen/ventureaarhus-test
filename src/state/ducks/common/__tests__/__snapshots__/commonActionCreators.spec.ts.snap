// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common action creators should properly create a change action for connection status using snapshot 1`] = `
Object {
  "payload": false,
  "type": "common/CHANGE_CONNECTION_STATUS",
}
`;
